lib/middleware/validate.js

```diff
@@ -0,0 +1,52 @@
+const { chain, unionWith, isEqual, size } = require('lodash');
+const validate = require('../validation');
+const { defaultValidators } = validate;
+
+const normaliseValidators = field => {
+  if (!field.validate) {
+    return field;
+  }
+  return {
+    ...field,
+    validate: field.validate.map(v => {
+      if (typeof v === 'string') {
+        return { type: v };
+      }
+      const type = Object.keys(v)[0];
+      return {
+        type,
+        params: v[type]
+      };
+    })
+  };
```

Comment:
This probably needs to handle the case where a validator is defined as the normalised form. i.e. if I set the validator as `{ type: 'maxlength': params: 20 }`

This will "normalise" that to `{ type: 'type', params: 'maxlength' }`

Reply:
good point 👍